### PR TITLE
seq_comm_mct code cleanup

### PR DESCRIPTION
**Pull request by jedwards4b**
Refactored to get rid of a lot of repeated code.   This will help with the multi_driver branch.
Also remove some un-needed esmf code.

Test suite: cesm2 prealpha tests on cheyenne
Test baseline: cesm2_0_alpha07c
Test namelist changes: 
Test status: bit for bit

Fixes
User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:   RobJ, SteveG.

### CODE REVIEW

File: src/drivers/mct/shr/seq_comm_mct.F90 [hunk 1]
```diff
@@ -917,27 +474,114 @@ subroutine seq_comm_init(Comm_in, nmlfile)
...
+    integer, intent(in) :: num_inst_comp
+    integer, intent(in) :: CPLID
+    integer, intent(inout) :: COMPID(num_inst_comp)
+    integer, intent(inout) :: CPLCOMPID(num_inst_comp)
+    integer, intent(inout) :: ALLCOMPID
+    integer, intent(inout) :: CPLALLCOMPID
+    integer, intent(inout) :: count
+    character(len=*), intent(in) :: name
+
+    character(len=*), parameter :: subname = "comp_pelayout"
```
Comment (gold2718) [inline src/drivers/mct/shr/seq_comm_mct.F90]: Why does subname not match the subroutine name?

File: src/drivers/mct/shr/seq_comm_mct.F90 [hunk 2]
```diff
@@ -917,27 +474,114 @@ subroutine seq_comm_init(Comm_in, nmlfile)
...
+       count = count + 1
+       CPLCOMPID(n) = count
+    enddo
+
+    if (mype == 0) then
+       !--- validation of inputs ---
+       ! rootpes >= 0
+       !! Determine the process layout
+       !!
+       !! We will assign atm_ntasks / num_inst_atm tasks to each atmosphere
```
Comment (gold2718) [inline src/drivers/mct/shr/seq_comm_mct.F90]: This comment should be generalized (not just for atm).

File: src/drivers/mct/shr/seq_comm_mct.F90 [hunk 3]
```diff
@@ -917,27 +474,114 @@ subroutine seq_comm_init(Comm_in, nmlfile)
...
+       !! instance.  (This may lead to unallocated tasks if atm_ntasks is
+       !! not an integer multiple of num_inst_atm.)
+
+       if (comp_rootpe < 0) then
+          write(logunit,*) trim(subname),' ERROR: rootpes must be >= 0'
+          call shr_sys_abort(trim(subname)//' ERROR: rootpes >= 0')
```
Comment (gold2718) [inline src/drivers/mct/shr/seq_comm_mct.F90]: Include `name` in error message so users know which component is bad.

File: src/drivers/mct/shr/seq_comm_mct.F90 [hunk 4]
```diff
@@ -917,27 +474,114 @@ subroutine seq_comm_init(Comm_in, nmlfile)
...
+       endif
+
+       if (trim(comp_layout) == trim(layout_concurrent)) then
+          comp_inst_tasks = comp_ntasks / num_inst_comp
+          droot = (comp_inst_tasks * comp_pestride)
+       elseif (trim(comp_layout) == trim(layout_sequential)) then
+          comp_inst_tasks = comp_ntasks
+          droot = 0
+       else
+          call shr_sys_abort(subname//' ERROR invalid comp_layout ')
```
Comment (gold2718) [inline src/drivers/mct/shr/seq_comm_mct.F90]: Include `name` in error message so users know which component is bad.

**Comment by jedwards4b** [comment]
@gold2718 thanks for the review, all comments incorporated.

File: src/drivers/mct/shr/seq_comm_mct.F90 [hunk 5]
```diff
@@ -917,27 +474,113 @@ subroutine seq_comm_init(Comm_in, nmlfile)
...
+       endif
+
+       if (trim(comp_layout) == trim(layout_concurrent)) then
+          comp_inst_tasks = comp_ntasks / num_inst_comp
+          droot = (comp_inst_tasks * comp_pestride)
+       elseif (trim(comp_layout) == trim(layout_sequential)) then
+          comp_inst_tasks = comp_ntasks
+          droot = 0
+       else
+          call shr_sys_abort(subname//' ERROR invalid comp_layout for component'//name)
```
Comment (gold2718) [inline src/drivers/mct/shr/seq_comm_mct.F90]: This is going to look funny without a space after component. Also, name should be trim(name) even though it is always being called with constants at the moment.

**Comment by jedwards4b** [comment]
fixed, added trim

**Comment by jedwards4b** [comment]
Found a problem, will need to rerun prealpha testing before this is ready to merge.

**Comment by jedwards4b** [comment]
Retested with cesm prealpha tests, all looks good.

**Comment by jedwards4b** [comment]
@rljacob this is ready and waiting for your review...

**Comment by rljacob** [comment]
Looks good.  I updated the description.

**Pull request merged**